File: java/2441. Largest Positive Integer That Exists With Its Negative.java
class Solution {
    public int findMaxK(int[] nums) {
        int answer = Integer.MIN_VALUE;
        Set<Integer> set = new HashSet<>();
        for (int i : nums) {
            set.add(i);
        }
        for (int i : nums) {
            if (i > 0 && set.contains(-i)) {
                answer = Math.max(answer, i);
            }
        }
        return answer == Integer.MIN_VALUE ? -1 : answer;
    }
}

File: python/2441. Largest Positive Integer That Exists With Its Negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        freq = [0] * 1001
        answer = -1
        for num in nums:
            if num < 0:
                freq[-num] += 1
        for num in nums:
            if num > 0 and freq[num] > 0 and answer < num:
                answer = num
        return answer
